Production_scripts/Create_costs/costs_district_nursing.R

```diff
@@ -0,0 +1,280 @@
+#####################################################
+# Costs - District Nursing
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input - DN_Costs.xlsx
+# Description - Create Lookup for District Nursing Costs.
+#
+# Download latest costs file from Cost Book website
+# http://www.isdscotland.org/Health-Topics/Finance/Costs/Detailed-Tables/index.asp
+#
+# Check and add costs to the Excel file 'DN_Costs.xlsx'. ##
+#
+# Extract numbers of contacts from the CHAD - District Nursing Datamart
+# using the query: DN-Contacts-Numbers-for-Costs.
+# This should be run/scheduled and downloaded as a .csv
+# Check the numbers in this file as some data completeness issues
+# may mean the numbers can't be used to create costs.
+#####################################################
+
+# Load packages #
+library(dplyr)
+library(tidyr)
+library(ggplot2)
+library(purrr)
+library(createslf)
+
+
+# Copy existing file ---------------------------------------
+
+# read data in
+current_file <- haven::read_sav(get_dn_costs_path())
+
+# Create a copy for comparison
+fs::file_copy(
+  get_dn_costs_path(),
+  get_dn_costs_path(update = latest_update())
+)
+
+# Read in cost workbook ---------------------------------------
+
+# latest year #
+latest_year <- "1920"
+
+## data ##
+dn_raw_costs <- readxl::read_excel(
+  fs::path(get_slf_dir(), "Costs", "DN_Costs.xlsx"),
+  .name_repair = janitor::make_clean_names
+) %>%
+  # change 1718 type to numeric - reads in as a character
+  mutate(across(ends_with("_cost"), as.numeric)) %>%
+  # pivot longer
+  pivot_longer(
+    ends_with("_cost"),
+    names_to = "year",
+    names_pattern = "(\\d{4})_cost",
+    values_to = "cost"
+  )
+
+# Read DN file extracted from BOXI -----------------------------
+
+# contacts
+dn_raw_contacts <- readr::read_csv(
+  fs::path(get_slf_dir(), "Costs", "DN-Contacts-Numbers-for-Costs.csv"),
+  name_repair = janitor::make_clean_names
+) %>%
+  # create year variable as fy
+  mutate(year = convert_year_to_fyyear(contact_financial_year)) %>%
```

Comment:
I get an error here but i can't figure out why. Seems to be a problem with the variable `contact_financial_year` which doesn't want to convert to a character with the new `check_year_format` function. I've checked the `check_year_format` function with various examples using e.g. 2019 as numeric and character, or 1920 as numeric or character and the function looks like it is working fine. However, using the variable `contact_financial_year` in the same way pulls through this error: 

`Error in format_error_bullets(x[-1]) : 
  nms %in% c("i", "x", "") are not all TRUE`

Do you get this as well? This line works totally fine if you convert `contact_financial_year` into a character before using convert_year_to_fyyear

Reply:
just re-ran and I do not get any error when running the script